### PR TITLE
Attempt to resolve: https://github.com/dask/dask/issues/6307

### PR DESCRIPTION
**Pull request by asmith26**
* Add function (that is called by derived_from) to update docstrings for methods containing the `split_out` parameter.
* Add `split_out` documentation to dataframe best practices.

- [ ] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: docs/source/dataframe-best-practices.rst [hunk 1]
```diff
@@ -31,6 +32,19 @@ want to switch to Pandas at this point.
    result = result.compute()                # Convert to Pandas dataframe
    result...                                # Continue working with Pandas
 
+
+
+By default groupby methods return an object with only 1 partition. This is to
```
Comment (TomAugspurger) [inline docs/source/dataframe-best-practices.rst:37]: This doesn't quite feel like the right section for this. The header here is "Reduce, then use Pandas" so the assumption is that the user wants an in-memory object back.

I think a new section at the end of `dataframe-design.rst` is best.

Comment (asmith26) [reply docs/source/dataframe-best-practices.rst:37]: Done in https://github.com/dask/dask/pull/6318/commits/edfc7a50f719a04d0ca25e1579b71a2fd933cfd7

**Comment by asmith26** [comment]
Thanks for the feedback @TomAugspurger

I agree, I've created a new section in dataframe-design.rst

Regarding the indentation problem, I've now noticed a few differences to how the online Dask doc looks and what I'm building locally (including for parts of the docs I haven't touched). Is there anywhere I can see how my changes are actually built for the website (e.g. I think they are built by maybe ReadTheDocs)?

**Comment by TomAugspurger** [comment]
Perhaps a difference in dependencies? The docs are built with
`docs/requirements-docs.txt`. I think you can view the build logs at
https://readthedocs.org/projects/dask/builds/11243345/ to verify.

**Comment by asmith26** [comment]
> Perhaps a difference in dependencies? The docs are built with `docs/requirements-docs.txt`. 

Thanks for the suggestion, I'm using these requirements.

> I think you can view the build logs at https://readthedocs.org/projects/dask/builds/11243345/ to verify.

Thanks. I had a thought and realized I could build the docs myself with readthedocs. Most things are looking better/as expected, including the new section in dataframe-design.rst: https://asmith26-demo.readthedocs.io/en/latest/dataframe-design.html#groupby

Unfortunately building with readthedocs has not fixed the indentation problem, e.g. [`drop_duplicates`](https://asmith26-demo.readthedocs.io/en/latest/dataframe-api.html#dask.dataframe.DataFrame.drop_duplicates) is misaligned, though some are OK (as I found locally) like [`size`](https://asmith26-demo.readthedocs.io/en/latest/dataframe-api.html#dask.dataframe.groupby.DataFrameGroupBy.size). The misalignment appears to be due to some of the original docs using `<blockquotes>` html tags whilst (I think the majority of) others use just standard `<p>` tags. E.g. for `drop_duplicates`:

```html
<blockquote>
<div><p>Return DataFrame with duplicate rows removed.</p>
<p>This docstring was copied from pandas.core.frame.DataFrame.drop_duplicates.</p>
<p>Some inconsistencies with the Dask version may exist.</p></div>
</blockquote>

<p>An explanation of the <cite>split_out</cite> parameter can be found <a class="reference internal" href="dataframe-design.html#dataframe-design-groupby"><span class="std std-ref">here</span></a>.</p>

<blockquote>
<div><p>Considering certain columns is optional. Indexes, including time indexes
are ignored.</p></div>
</blockquote>
```

Consequently, I feel there are a few ways to proceed:
1. Just add the new section in the dataframe-design.rst (i.e. not update the API docs).
2. Try to understand why some of the (I think pandas) docs are creating `<blockquote>` html tags (I'm pretty flummoxed by this though).
3. Commit this and don't worry about the misalignment of the doc (not as pretty as it could be, but does inform the user).

I think for simplicity, I'm happy to go with 1. What are your thoughts?

**Comment by TomAugspurger** [comment]
I think 1 sounds good as well.

**Comment by asmith26** [comment]
Thanks for letting me know. I think my latest push completes this now then.

**Comment by TomAugspurger** [comment]
Thanks @asmith26!

**Pull request merged**